TACC/Core-CMS
# Quick: Un-Deprecate Bootstrap Picture

## Overview

Do **not** warn users about using Picture plugin (and clarify when to use).

_The default responsive behavior of Bootstrap's image is desirable for some content._

<details><summary>Background</summary>

1. During #278, the warning was added.
2. Since then, I have accepted Bootstrap as part of the CMS.
3. During [ECEP-113](https://jira.tacc.utexas.edu/browse/ECEP-113), I realized the use case for Bootstrap picture.

</details>

## Related

* expected by #466

## Changes

- Remove deprecation icon and text.
- Change name of plugin to reflect intended usage.

## Testing

You can test on [dev.cep](https://dev.cep.tacc.utexas.edu/test/quick-un-deprecate-bootstrap-picture/) or locally.

1. Create page.
4. Add image via Bootstrap Picture plugin.
    - Verify **no** ⚠️ icon **nor** deprecation text.
    - Verify new name "Picture / Image (Responsive)"
5. Add image via Django Image plugin.
6. Both images should load fine.

## Screenshots

<details open><summary>What is the Change for Users?</summary>

| | Before | After |
| - | - | - |
| Search for Plugin | ![Before - Plugin Search](https://user-images.githubusercontent.com/62723358/165321074-a59302dc-c3c8-4a3c-80e3-b17fe6e03579.png) | ![After - Plugin Search](https://user-images.githubusercontent.com/62723358/165321068-3e19204f-39e5-4aac-8aa6-0f05bd79ea60.png) |
| Add the Plugin | ![Before - Plugin Create](https://user-images.githubusercontent.com/62723358/165321071-1461b13f-40ad-4f5b-9fe7-fb8dafaacc9c.png) | ![After - Plugin Create](https://user-images.githubusercontent.com/62723358/165321061-10f40937-7dd0-4d46-a480-4131702ae813.png) |

</details>

<details><summary>Why Support Bootstrap Picture?</summary>

1. Bootstrap Picture applies the class `img-fluid` by default, and has a clear setting—a checkbox labeled "Responsive".
    Users don't know they usually want this. For devs, it's easier to allow Bootstrap Picture than customize the generic Picture plugin.

    | With "Responsive" Setting / `img-fluid` Class | Sans "Responsive" Setting / `img-fluid` Class |
    | - | - |
    | ![With img-fluid Class](https://user-images.githubusercontent.com/62723358/165324969-74a5d3cf-796d-4a1b-9c81-54f36d6085ca.png) | ![Sans img-fluid Class](https://user-images.githubusercontent.com/62723358/165324965-3e74cf95-215d-464b-ab7e-ac5823baa58a.png) |
    | ✅  image fits in text/content box | ❌  image escapes text/content box |

3. Django "Image" (above) is not responsive, so it can exceed container width (left). Bootstrap "Picture" (below) is responsive by default, so it will never exceed container width (left). If container is wider than images, neither image exceeds container width (right).

    | 1200px | 991px |
    | - | - |
    | ![991px](https://user-images.githubusercontent.com/62723358/160912899-4a11af9b-1a84-432f-b891-2305c0a38b14.png) | ![1200px](https://user-images.githubusercontent.com/62723358/160912886-c6df7d04-f35c-496c-ad4f-e85bdcff9362.png) |
    | window width < image width<br />Bootstrap "Picture" is only as wide as container | window width > images width, same size<br />image is only as big as intrinsic size |

</summary>
